feat: support GWAPI frontend validation

kinolaev opened the pull request:
**Please provide a description of this PR:**

This PR adds support for GWAPI frontend validation. Requires https://github.com/istio/istio/pull/56549 to be merged first.

howardjohn reviewed (commented):
can we add a unit test under pilot/pkg/config/kube/gateway/testdata? This should cover the translation. Can maybe add to pilot/pkg/config/kube/gateway/testdata/tls.yaml

kinolaev commented:
I've updated the PR, cacert is referenced with the new `caCertCredentialName` field. I've also added tests (thanks for the hint, @howardjohn), release note and docs https://github.com/istio/istio.io/pull/16739. I think this PR is ready for review now.

kinolaev commented:
@howardjohn , @keithmattix , could you please add /ok-to-test label?

zirain commented:
/ok-to-test

keithmattix reviewed (commented):
Thanks for sticking with this! Looks generally good, though I think we're past code freeze for 1.27

kinolaev commented on pilot/pkg/config/kube/gateway/testdata/reference-policy-tls.yaml.golden:
```diff
@@ -17,8 +17,9 @@ spec:
+      caCertCredentialName: configmap://default/auth-cert-cacert
       credentialName: kubernetes-gateway://cert/cert
-      mode: SIMPLE
+      mode: MUTUAL
```
yes, because I've added `frontendValidation` with a ConfigMap. That way we check that ReferenceGrants works for both types (Secret and ConfigMap)

istio-testing commented:
@kinolaev: The following tests **failed**, say `/retest` to rerun all failed tests or `/retest-required` to rerun all mandatory failed tests:

Test name | Commit | Details | Required | Rerun command
--- | --- | --- | --- | ---
integ-ambient-owned-cni_istio | 4c52bace1e1e69c9f3051dc83a421a96a6242231 | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/57183/integ-ambient-owned-cni_istio/1952283584794464256) | false | `/test integ-ambient-owned-cni`
integ-pilot-multicluster_istio | cca5e4c21442d1907eb0ce78ab9fb99903f25e6e | [link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio/57183/integ-pilot-multicluster_istio/1953747814550867968) | true | `/test integ-pilot-multicluster`

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes-sigs/prow](https://github.com/kubernetes-sigs/prow/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

kinolaev commented:
The PR is rebased to the master branch to make all checks green and get some more attention from istio maintainers)

howardjohn reviewed (approved):
thanks!
